src/components/FeatureViews/Accounts/TabComponents/InstructorSchedule.js

```diff
@@ -1,132 +1,130 @@
-					// must add one day to include the end specified by user
-					if (all_day) {
-						endDate.setDate(endDate.getDate() + 1);
-					}
-					return {
-						allDay: all_day,
-						end: endDate,
-						start,
-						title: description || "Out of Office",
-					};
-				}
-				)
-				: [],
-		[instructor]
-	);
 
-	const allDayOOO = useMemo(() => OOO.some(({allDay}) => allDay), [OOO]);
-
-	const hoursWorked = useMemo(
-		() =>
-			Math.round(
-				Object.values(sessions[instructorID] || []).reduce((hours, session) => {
-					const start = new Date(session.start_datetime);
-					const end = new Date(session.end_datetime);
-					if (start > Date.now()) {
-						return hours;
-					} else if (end < Date.now()) {
-						return hours + toHours(end - start);
-					}
-					return hours + toHours(end - Date.now());
-				}, 0) * 100
-			) / 100,
-		[sessions, instructorID]
-	);
+	const { loading, error, data } = useQuery(GET_INSTRUCTOR_INFO, {
+		variables: {instructorID}
+	})
 
-	const instructorBusinessHours = useMemo(
-		() =>
-			instructor
-				? Object.values(instructor.schedule.work_hours).map(
-				({day, start, end}) => ({
-					daysOfWeek: [day],
-					endTime: end,
-					startTime: start,
-				})
-				)
-				: [],
-		[instructor]
-	);
+	if (loading ) return <Loading small/>;
 
-	if (
-		teachingSessions.length === 0 &&
-		instructorBusinessHours.length === 0 &&
-		OOO.length === 0
-	) {
-		if (
-			hooks.isLoading(
-				tutoringEnrollmentStatus,
-				OOOstatus,
-				classEnrollmentStatus,
-				availabilityStatus
-			)
-		) {
-			return <Loading/>;
-		} else if (
-			hooks.isFail(
-				tutoringEnrollmentStatus,
-				OOOstatus,
-				classEnrollmentStatus,
-				availabilityStatus
-			)
-		) {
-			return "Unable to load schedule!";
+	if (error ) return  <Typography>There was an error {error.message}</Typography>;
+
+	const { user, user : { instructor } } = data.instructor;
+
+	const today = moment();
+
+	const hoursWorkedThisMonth = parseInt(instructor.sessionSet.reduce((hours, session) => {
+		const start = moment(session.startDatetime);
+		const end = moment(session.endDatetime);
+
+		// Ignore sessions in the future
+		if (today.diff(start) < 0) {
+			return hours;
+		}
+		// If session is in the same month add hours
+		else if (start.month() === today.month() &&
+			start.year() === today.year()) {
+			return hours + toHours(end.diff(start));
+		} else {
+			return hours;
 		}
-	}
 
+	}, 0));
+
+	const instructorBusinessHours = instructor.instructoravailabilitySet.map(({dayOfWeek, startTime, endTime}) => ({
+		dayOfWeek: [dayOfWeek],
+		startTime, 
+		endTime
+	}))
+
+	const teachingSessions = instructor.sessionSet.map(({endDatetime, startDatetime, title}) => ({
+		title: title,
+		start: new Date(startDatetime),
+		end: new Date(endDatetime)
+	}));
+
+	let allDayOOO = false;
+	// Out Of Office
+	const OOOEvents = instructor.instructoroutofofficeSet.map(({startDatetime, endDatetime, description}) => {
+		let start = new Date(startDatetime);
```

Comment:
Same as above

Reply:
I talked with Daniel about this. Turns out FullCalendar requires a plugin to use moment dates, so Daniel said it would be fine to use moment to compare times and use Date objects to use in the FullCalendar events